daemon/src/main/java/bisq/daemon/grpc/GrpcMoneroConnectionsService.java

```diff
@@ -0,0 +1,316 @@
+import static bisq.proto.grpc.MoneroConnectionsGrpc.*;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+
+
+import bisq.daemon.grpc.interceptor.CallRateMeteringInterceptor;
+import bisq.daemon.grpc.interceptor.GrpcCallRateMeter;
+
+@Slf4j
+class GrpcMoneroConnectionsService extends MoneroConnectionsImplBase {
+
+    private final CoreApi coreApi;
+    private final GrpcExceptionHandler exceptionHandler;
+
+    @Inject
+    public GrpcMoneroConnectionsService(CoreApi coreApi, GrpcExceptionHandler exceptionHandler) {
+        this.coreApi = coreApi;
+        this.exceptionHandler = exceptionHandler;
+    }
+
+    @Override
+    public void addConnection(AddConnectionRequest request,
+                              StreamObserver<AddConnectionReply> responseObserver) {
+        handleRequest(responseObserver, () -> {
+            coreApi.addMoneroConnection(toInternalUriConnection(request.getConnection()));
+            return AddConnectionReply.newBuilder().build();
+        });
+    }
+
+    @Override
+    public void removeConnection(RemoveConnectionRequest request,
+                                 StreamObserver<RemoveConnectionReply> responseObserver) {
+        handleRequest(responseObserver, () -> {
+            coreApi.removeMoneroConnection(validateUri(request.getUri()));
+            return RemoveConnectionReply.newBuilder().build();
+        });
+    }
+
+    @Override
+    public void getConnection(GetConnectionRequest request,
+                              StreamObserver<GetConnectionReply> responseObserver) {
+        handleRequest(responseObserver, () -> {
+            bisq.proto.grpc.UriConnection replyConnection = toGrpcUriConnection(coreApi.getMoneroConnection());
+            return GetConnectionReply.newBuilder().setConnection(replyConnection).build();
+        });
+    }
+
+    @Override
+    public void getConnections(GetConnectionsRequest request,
+                               StreamObserver<GetConnectionsReply> responseObserver) {
+        handleRequest(responseObserver, () -> {
+            List<UriConnection> connections = coreApi.getMoneroConnections();
+            List<bisq.proto.grpc.UriConnection> replyConnections = connections.stream()
+                    .map(GrpcMoneroConnectionsService::toGrpcUriConnection).collect(Collectors.toList());
+            return GetConnectionsReply.newBuilder().addAllConnections(replyConnections).build();
+        });
+    }
+
+    @Override
+    public void setConnection(SetConnectionRequest request,
+                              StreamObserver<SetConnectionReply> responseObserver) {
+        handleRequest(responseObserver, () -> {
+            coreApi.setMoneroConnection(validateUri(request.getUri()));
+            return SetConnectionReply.newBuilder().build();
+        });
+    }
+
+    @Override
+    public void extendedSetConnection(ExtendedSetConnectionRequest request,
```

Comment:
Rather than the grpc api supporting `setConnection(string)` and `extendedSetConnection(UriConnection)`, it can support only one method, `setConnection(uriConnection)`. The client HavenoDaemon.ts class can locally support `setConnection(string)` which constructs a `UriConnection` and invokes the single common grpc method, as it is a superset.

This should only affect the grpc classes, not CoreApi, MoneroConnectionsManager, etc. Those classes can keep the two methods, `setConnection(string)` and `setConnection(uriConnection)`.